### ISSUE TITLE
Open Beta v8.1 | Please help testing and hardening the upcoming release

### ISSUE THREAD

**Issue by MichaIng** [MichaIng/DietPi#5218]
RC version | v8.1.2
---------- | -------
v8.1.0 PR | #5215
v8.1.1 PR | #5226
v8.1.2 PR | #5230
Raw changelog | https://github.com/MichaIng/DietPi/blob/beta/CHANGELOG.txt
Code changes | https://github.com/MichaIng/DietPi/compare/master...beta
Release date | **2022-02-05**

#### Related/solved issues: https://github.com/MichaIng/DietPi/issues?q=is%3Aissue+milestone%3Av8.1
____
### Beta v8.1.2
_(2022-02-05)_

### SBC support
- Odroid N2 | Support for Odroid N2 with modern mainline Linux images has been added, including proper boot configuration, serial console selection, HDMI and 3.5mm audio selection and more. Sadly existing images cannot be migrated, a new image is available for testing: https://github.com/MichaIng/DietPi/issues/5039

### Changes
- DietPi-Benchmark | During RAM I/O benchmarks, the CPU governor is now changed to "performance" so that the result is not affected by possible CPU frequency scaling but the raw/max RAM I/O performance is measured. A significant effect of this has been recognised on Odroid N2 images with modern mainline kernel. Since the CPU benchmark is done first, this has no effect on its score and CPU temperature results.
- DietPi-Drive_Manager | Btrfs subvolume mounts are now preserved in /etc/fstab. Many thanks to @laddde for implementing this feature: https://github.com/MichaIng/DietPi/pull/5176
- DietPi-Drive_Manager | The Samba mount dialogues do now indicate that a hostname can be entered instead of a server IP and a shared folder path instead of a share name only.
- DietPi-Backup | A notification is not printed when rsync dry-run to obtain required disk space is starting, since this can take a while. Many thanks to @whisdol for reporting the the impression of a hanging script: https://github.com/MichaIng/DietPi/issues/5209
- DietPi-Software | iptables is now automatically configured to use either modern nf_tables API or legacy API, depending on whether nf_tables is supported by the kernel. This is helpful for Bullseye images with older kernel versions, where iptables is by default configured to use nf_tables, which fails.
- DietPi-Software | iptables rules are now applied for IPv6 as well in all cases (for VPN servers, WiFi and Tor hotspots) via ip6tables. This does not depend on whether IPv6 is actually enabled for interfaces, but enables general IPv6 support for affected software options. Many thanks to @vielka442 for reporting missing IPv6 support for Tor Hotspot: https://github.com/MichaIng/DietPi/issues/5038
- DietPi-Software | Jellyfin: For new installs, the default web interface port has been changed from 8096 to 8097 to resolve a port conflict with Emby.
- DietPi-Software | Docker/K3s: The memory cgroup is now enabled as well on RPi and our new Odroid N2 image, when Docker or K3s is installed, to allow container memory usage monitoring. Many thanks to @isarrider for this request: https://github.com/MichaIng/DietPi/issues/5225
- DietPi-Software | Raspotify: This software option has been disabled on Debian Buster systems but enabled for ARMv8 and x86_64 systems, to match the current state of the official Raspotify APT repository.

### Fixes
- DietPi-Services | Resolved an issue where service edits and process tool settings were not effective as they were stored to the wrong file path. Many thanks to @TopFord and @MicDG for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?p=41509#p41509
- DietPi-LetsEncrypt | Resolved an issue where Certbot failed on fresh Apache installs since the 000-default.conf vhost by default does not contain any directive anymore. The ServerName directive with the entered domain name is now added to 000-default.conf instead of apache2.conf, with the benefit that it overrides the global ServerName directive in dietpi.conf, which uses the local IP address only.
- DietPi-Software/DietPi-AutoStart | Chromium: Resolved an issue where the Chromium kiosk autostart option failed on all non-RPi systems. 
- DietPi-Software | FuguHub: Resolved an issue where the uninstall failed as the service was not stopped and removed as expected, before attempting to remove the service user.
- DietPi-Software | PHP: Resolved an issue where the PATH variable was not available for PHP applications with Lighttpd and Nginx webservers. It is now set to the static value "/usr/local/bin:/usr/bin:/bin", the default PATH excluding super user executables "sbin". Many thanks to @shanew1694 for reporting a related issue with rTorrent: https://github.com/MichaIng/DietPi/issues/5161#issuecomment-1013381362
- DietPi-Software | MineOS: Resolved an issue where the install failed on Bullseye: https://github.com/MichaIng/DietPi/issues/5181
- DietPi-Software | Cuberite: Resolved an issue where the server failed to start on ARMv8 systems: https://github.com/MichaIng/DietPi/issues/5180
- DietPi-Software | MATE: Resolved an issue where the install failed when LXDE is installed already. Many thanks to @n0valis for reporting this issue: https://github.com/MichaIng/DietPi/issues/5214
- DietPi-Software | Ampache: Resolved an issue on Bullseye systems where Ampache v5 installs/reinstalls did not work because of a changed webroot directory: https://github.com/MichaIng/DietPi/pull/5205
- DietPi-Software | OpenBazaar: Since the original project has been taken down, we migrated to a fork which keeps up the marketplace for the community: https://github.com/MichaIng/DietPi/issues/5213
- DietPi-Software | X.Org X Server: Resolved an issue where VMs without KMS/DRM support failed to start the X server, e.g. Hyper-V. The classic framebuffer DDX is now installed when no KMS/DRM support is detected on VMs.
- DietPi-Software | Box86/Box64: Resolved an issue where the install failed when a kernel upgrade has just been done so that the "binfmt_misc" kernel module cannot be found for the loaded kernel.
- DietPi-Software | Roon Extension Manager: It has been disabled for ARMv6 RPi systems where it cannot run currently. Many thanks to @JanKoudijs for applying this fix: https://github.com/MichaIng/DietPi/pull/5231
- DietPi-Software | RPi-Monitor: Resolved an issue where the CPU temperature was not shown. Many thanks to `@KamikazeePL` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=10001
- DietPi-Software | Home Assistant: Resolved an issue with the latest Home Assistant version where restarting it via web interface made it remaining stopped. The fix is applied to all instances on DietPi update. Many thanks to `@jchh` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=10025

**Comment by Joulinar** [comment]
@MichaIng 
it is not directly related to the Beta but might be something we could/should include. I noticed that desktop systems are not able boot on HyperV VM systems. Because video package `xserver-xorg-video-fbdev` is missing. I know we only install `xserver-xorg-video-vmware` in general on all VM's (HW ID 20)

https://github.com/MichaIng/DietPi/blob/fa396c402ca1dfa902de2e074d8e50f97b60ef6d/dietpi/dietpi-software#L2661-L2662

**Comment by MichaIng** [comment]
Hyper-V does not support KMS/DRM?
```sh
ls -l /dev/dri
```
We could check whether any `/dev/dri/card[0-9]` exists and based on this decide whether to install `xserver-xorg-video-fbdev` or `xserver-xorg-video-vmware`.

**Comment by Joulinar** [comment]
there is no `/dev/dri` on the HyperV VM

```
root@DietPi:~# ls -l /dev/dri
ls: cannot access '/dev/dri': No such file or directory
root@DietPi:~#
```

**Comment by MichaIng** [comment]
Solved with: https://github.com/MichaIng/DietPi/commit/a6340fa

**Comment by MichaIng** [comment]
#### DietPi Beta v8.1.1 has been merged: #5226

**Comment by Joulinar** [comment]
I tested desktop install on HyperV which seems to be working fine now. Correct package `server-xorg-video-fbdev` has been selected 👍 

```
 DietPi-Software
─────────────────────────────────────────────────────
 Step: Installing X.Org X Server: aka X11 - X Window System implementation

[  OK  ] DietPi-Software | cd /tmp/DietPi-Software
[  OK  ] DietPi-Software | Checking URL: https://raw.githubusercontent.com/MichaIng/DietPi/beta/.conf/dps_6/98-dietpi-disable_dpms.conf
[ INFO ] DietPi-Software | G_THREAD_START_0 | curl -sSfL https://raw.githubusercontent.com/MichaIng/DietPi/beta/.conf/dps_6/98-dietpi-disable_dpms.conf -o 98-dietpi-disable_dpms.conf
[ INFO ] DietPi-Software | APT install for: xserver-xorg-core xserver-xorg-input-libinput xinit dbus-x11 xfonts-base x11-xserver-utils x11-utils libgl1-mesa-dri mesa-utils mesa-utils-extra xserver-xorg-video-fbdev, please wait...
```

Desktop is correctly started afterwards.

**Comment by MichaIng** [comment]
Great, thanks for testing 👍🙂

**Comment by Joulinar** [comment]
@MichaIng 
Just as information. There seems to be an issue to restart HA from GUI as the service is going to be killed by own application. Similar to what we had on Webmin. Unfortunately the same Workaround we applied there, is not working on HA. Maybe you can have a look if you find a better solution. https://dietpi.com/phpbb/viewtopic.php?p=42859#p42859

**Comment by MichaIng** [comment]
A very recent change indeed. Here it is: https://github.com/home-assistant/core/issues/65450
Hence:
```sh
echo -e '[Service]\nRestartForceExitStatus=100' > /etc/systemd/system/home-assistant.service.d/dietpi-services_edit.conf
systemctl daemon-reload
systemctl restart home-assistent
```
I'll add it to `dietpi-software` as well.

**Comment by MichaIng** [comment]
#### DietPi Beta v8.1.2 has been merged: #5230

After a last review, this will be merged to stable.

**Comment by MichaIng** [comment]
#### DietPi v8.1 has been released: #5229

Many thanks to all testers ❤️
